pipeline/routing/ppr.py

```diff
@@ -32,7 +34,128 @@ def insertPathsElementsRef(cur, edges, path_counter):
+    # id(id,level):   ----------------- (id_2,NULL) --------- (id_3,1) ----------- (id_3,-1) --------- (id_4,NULL) --------------------
+    
+    # Special cases:
+    
+    # Elevators:
+    # Their 'osm_way_id', 'from_node_osm_id' and 'to_node_osm_id' are the same. One elevator can have multiple levels, and therefore multiple access spaces.
+    # So the elevators access spaces are identified by the level of the previous edge when stepping into the elevator,
+    # and the level of the current edge when stepping out of the elevator.
+    
+    # Escalators:
+    # Their level is dependent on the direction of the path. So the access spaces are identified by the level of the previous edge when going into the escalator,
+    # and the level of the current edge when going out of the escalator.
+    
+    # Stairs:
+    # They always have the same level, regardless of the direction. So the access spaces are identified by the level of the previous edge when going into the stairs,
+    # and the level of the current edge when going out of the stairs.
+    
+    for edge in edgeIter:
+        edge_type = edge["edge_type"]
+        previous_edge_type = previous_edge["edge_type"]
+        street_type = edge["street_type"]
+        previous_street_type = previous_edge["street_type"]
+        path = edge["path"]
+
+        if( # 1) transition from one edge_type to another
+            edge_type != previous_edge_type and
+            (edge_type in special_edge_types or previous_edge_type in special_edge_types)
+            )\
+            or( # 2) transition from one street_type to another
+                street_type != previous_street_type and
+                (street_type in special_street_types or previous_street_type in special_street_types)
+            ):
+            # special cases:
+            if edge_type == "elevator" or street_type == "stairs" or street_type == "escalator":
+                # going into the elevator/stairs/escalator: use level from the previous edge
+                # this might fail if two special cases are directly connected (e.g. escalator to stairs)
+                current_level = previous_edge["level"]
+            else:
+                # normal case: use current level
+                current_level = edge["level"]
+            
+            # create unique id for the access space, that will be filled into the 'IFOPT' column
+            # 'STOP_PLACE'_'OSM_NODE_ID':'LEVEL_IF_EXISTS'
+            ifopt = str(relation_id) + "_" + str(edge["from_node_osm_id"]) + ":" + (str(current_level) if current_level != None else "")
+            
+            insertAccessSpaces(cur, edge["from_node_osm_id"], current_level, ifopt, None, edge["path"][0])
+            
+            # insert pathLink into database
+            insertPathsLinks(cur, pathLink, id_from, ifopt)
+            id_from = ifopt
+        
+        # append edge to pathLink
+        if not pathLink:
+            pathLink = [path[0], path[1]]
+        else:
```

Comment:
Couldn't we spare this and instead init it like so `pathLink = [firstEdge["path"][0]]`

and then only add the last point:


            # insert pathLink into database
            insertPathsLinks(cur, pathLink, id_from, ifopt)
            # reset path and insert last point
            pathLink.clear() # or alternatively pathLink = [path[1]] and then handle the append differently
            id_from = ifopt

         # only append the second node of the path, because the first node is the same as the last node of the previous path
         pathLink.append(path[1])